compiler/control/OMROptions.cpp

```diff
@@ -3420,95 +3443,42 @@ OMR::Options::processOption(
-      //
-      opt = table+i;
-      if (!opt->length)
-         opt->length = strlen(opt->name);
-      cmp = strnicmp_ignore_locale(option, opt->name, opt->length);
-
-      // Some Z language environments transform:
-      //    '=' to '('
-      //    ':' to ')'
-      //    '_' to '\'
-      //
-      // Ensure, for example, 'log@lf' is the same as 'log=lf'
-      //
-      if (opt->name[opt->length - 1] == '=' &&
-          strlen(option) >= opt->length &&
-          option[opt->length - 1] == '@' &&
-          strnicmp_ignore_locale(option, opt->name, opt->length - 1) == 0)
-         {
-         cmp = 0;
-         }
-
-      if (cmp < 0)
-         {
-         if (i == low)
-            return startOption; // Not found in this table
-         high = i-1;
-         i = (i+low)/2;
-         }
-      else if (cmp > 0)
-         {
-         if (i == high)
-            return startOption; // Not found in this table
-         low = i+1;
-         i = (i+high+1)/2;
-         }
-      else
-         {
-         // Matching entry found
-         //
-         break;
-         }
+      if (!i->length)
+         i->length = strlen(i->name);
       }
 
-   // See if there is a better (longer) match with a subsequent entry
-   //
-   for (i++ ; i < numEntries; i++)
-      {
-      int32_t len = opt->length;
-      if (!option[len] || option[len] == ',' || option[len] == ')')
-         break;
-      TR::OptionTable *next = table+i;
-      if (!next->length)
-         next->length = strlen(next->name);
-      if (next->length <= len || strnicmp_ignore_locale(opt->name, next->name, opt->length) != 0)
-         {
-         // Stop if this entry can't be an extension
-         //
-         break;
-         }
-      cmp = strnicmp_ignore_locale(option, next->name, next->length);
+   // Find the entry for this option string using a binary search
 
-      if (next->name[next->length - 1] == '=' &&
-          strlen(option) >= next->length &&
-          option[next->length - 1] == '@' &&
-          strnicmp_ignore_locale(option, next->name, next->length - 1) == 0)
-         {
-         cmp = 0;
-         }
-      if (cmp < 0)
-         {
-         // Stop if we've gone past any possible match
-         //
-         break;
-         }
-      if (cmp == 0)
-         {
-         // This is a better match
-         //
-         opt = next;
-         }
+   // Create an object for the option to find in the table and set attributes
+   TR::OptionTable optionToFind = TR::OptionTable();
+   optionToFind.name = startOption;
+   optionToFind.length = strlen(optionToFind.name);
+
+   // Since STL binary search requires total ordering, there need to be a way to differentiate
+   // between the option to find and an option in the table. The msg field is used here to
+   // specify which option is the option to find. Note that this is not normal usage of the
+   // msg field.
+   optionToFind.msg = "optionToFind";
```

Comment:
While a clever way to leverage what's already in the OptionTable struct without adding anything new, I would prefer adding a `bool` to indicate this special property to make it crystal clear what the intent is.  It also places an undocumented dependency on an actual option message.  It is highly unlikely that this would ever cause a problem but we're leaving the possibility open.